src/compiler/core.ts

```diff
@@ -1488,6 +1488,119 @@ namespace ts {
         return createMultiMap() as UnderscoreEscapedMultiMap<T>;
     }
 
+    export function createSet<TElement, THash = number>(getHashCode: (element: TElement) => THash, equals: EqualityComparer<TElement>): Set<TElement> {

```

Comment:
if this stays in `src/compiler`, I think it should have a longer name to distinguish it from the built-in set, something that reflects that you can customise the equality predicate.

Reply:
Offline, we decided that an explanatory comment would suffice.